streams/src/main/java/org/apache/kafka/streams/processor/internals/TaskManager.java

```diff
@@ -1750,6 +1761,7 @@ Map<TaskId, Task> allTasks() {
         if (stateUpdater != null) {
             final Map<TaskId, Task> ret = stateUpdater.getTasks().stream().collect(Collectors.toMap(Task::id, x -> x));
             ret.putAll(tasks.allTasksPerId());
+            ret.putAll(tasks.pendingTasksToInit().stream().collect(Collectors.toMap(Task::id, x -> x)));
```

Comment:
@cadonna Could you please take a look at https://issues.apache.org/jira/browse/KAFKA-16774?

It seems this change could make StreamThreadTest#shouldCloseAllTaskProducersOnCloseIfEosEnabled failed.